src/cpp/shared/town/town.cpp

```diff
@@ -14,6 +14,15 @@
 
 #include<sstream>
 
+unsigned long gTownEligibleBuildMask[] = {
+  1073266591,
```

Comment:
These are bit masks. At the very least, they should be given in hexadecimal. Or possible in a form like ~0 & ~(1 << BUILDING_SHIPYARD) & ~(1 << BUILDING_DWELLING_3_UPGRADE), which is e.g.: the mask indicating that any building save the shipyard and the upgraded dwelling 3 may be built. (Unsure about parens.)

(If you do refactor this, you'll probably want to locally define some abbreviations.)

Reply:
I'm not sure how you were able to tell what the bit masks were composed of, but I'll change it to Hexadecimal for now.